### PR TITLE
ARPS first draft of water classification demo

### PR DESCRIPTION
**Pull request by olepinard**

### CODE REVIEW

**Comment by tbarsballe** [comment]
First, some general comments:
All the actual code looked simple and easy to follow - I don't think you need any changes there.
The overall workflow also made sense. Pretty much all my comments come down to text/description updates.

Notebook diffs are a challenge to comment on, so I'm going to leave specific feedback in this one comment. Everything should follow more or less sequentially:

All the notebooks in this repository expect only the packages installed in the docker image to be present (listed in [this requirements](https://github.com/planetlabs/notebooks/blob/master/planet-notebook-docker/requirements.txt) file). This does not currently include the `sentinelhub` package. Either update the notebook to install it via a `!pip install`, or update the docker file (I think in this case the latter makes more sense).

Given that the version of the notebook committed here is going to be the main reference version, it'd be nice to not have runtime warnings, i.e. 
Screenshot 2024-09-03 at 11 09 58 AM

These look more like empty headings, and should be reformatted to a bulleted list or similar.
Screenshot 2024-09-03 at 11 14 03 AM

I'd also suggest updating the steps to match the actual heading names later on (e.g. "Visualize RGB), or adding sub-steps that do (or alternatively, updating the headings to match the steps)

> In order to visualize imagery from Sentinel Hub we need to use evalscripts

I'd either link to the [SH reference on evalscripts](https://docs.sentinel-hub.com/api/latest/evalscript/) or briefly describe them here (i.e. mention they use JavaScript syntax, etc.)

> the API will return values between 0-255 and the ARPS data is a SR reflectance product that spans approximately 0-10,000

What's "the API" in this case? (I assume Processing - mention it inline). Also, _technically_ the 0-255 range only applies when using sampleType: UINT8; UINT32 is 0-65535 and FLOAT32 uses 0-1
**Edit**: I see you mention both these in the next section, can probably leave off the sample type description till then, but still mention the specific API here.


> Using the Processign API we are not just bound to rgb images
Fix spelling. Also, uppercase for RGB.

> You will notice that the NDWI values are much lower then with traditional Planet Scope imagery, that is normal and consistent across the ARPS dataset as a result of the harmonizing alogrithm.

I wouldn't necessarily assume that everyone who'll be using this notebook has examined NDWI in PlanetScope, perhaps switch "You will notice" to "You may notice". You could also mention what data products this NDWI value is more similar to and/or what ARPS is harmonized to.
Also - spelling/grammer, switch "harmonizing alogrithm" to "harmonization algorithm"

> (FILL IN)

I assume this is a placeholder.

```

{'data': [{'interval': {'from': '2022-04-30T12:00:00Z',
    'to': '2022-05-01T12:00:00Z'},
   'outputs': {'default': {'bands': {'B0': {'stats': {'min': -0.8281406760215759,
        'max': 0.3854166567325592,
        'mean': -0.6879543394116271,
        'stDev': 0.07646239025998604,
        'sampleCount': 558000,
        'noDataCount': 0}}}}}},
...
```
The statistical API output here adds a ton of length to the notebook. It is nice to show the structure though... not sure I've got any actionable feedback here, but it is a bit unwieldy as-is.

> Now we have a dataset but there are some dates that have NaN values throughout. Those are days where another region of the tile has valid imagery but not the area that we are looking at. Since ARPS is a tiled product the whole 24 km X 24 km tile is generated so Sentinel Hub will detect data that day even though its all NaN. On some days part of the image will be but not the entirety this is a problem if we are trying to determine the mean ndwi value of the AOI as the available imagery might have more or less lake to land ratio.

This is a bit run-on and hard to follow - took me a couple reads to get what it was trying to say. Consider rewriting.

Screenshot 2024-09-03 at 11 40 48 AM
A brief description of the implications of this plot would be nice. That'd also tie in nicely to the next section.

The header "**Image Classification**" should maybe be adjusted. While it's not wrong per se, I think most people these days think AI/ML when the term "Image Classification" is used.


The very last cell of the notebook is empty, and could be removed.

**Comment by strixcuriosus** [comment]
> If this is your first time using the Sentinel Hub APIs, please [check configuration instructions](https://sentinelhub-py.readthedocs.io/en/latest/configure.html) about how to set up your Sentinel Hub credentials.

It is also helpful to link to the place where you create the SH credentials, which is currently here: https://apps.sentinel-hub.com/dashboard/#/account/settings

and to show how to use them inline in the case where the user does not have an existing SH profile file set up and doesn't want to be bothered before forging ahead with this demo, e.g.
```
config = SHConfig()
if not config.sh_client_id:
    config.sh_client_id = '' # CHANGE THIS
    config.sh_client_secret = '' # CHANGE THIS

assert config.sh_client_id and config.sh_client_secret
```

**Comment by strixcuriosus** [comment]
Related to the previous comment on SHConfig, can you make get_stats reuse the config set up at the beginning of the notebook rather than reinitializing.

In other words, can you do something like this:
```
def get_stats(collection_id, eval, config):
     # run stats request
```
instead of this
```
def get_stats(collection_id, eval):
    config = SHConfig()
    # run stats request
```

**Comment by strixcuriosus** [comment]
Awesome demo! It's great to see some ARPS examples with visualization and stats workflows. I was able to run through all of it with only a few minor bumps related to SHConfig (mentioned in above comments).

As a future enhancement, it could be helpful to parallelize requests for the slower stats by month section. Currently it is waiting for each month to complete before starting the next.

Thanks for starting this example. I look forward to adding more ARPS analysis demos here.

**Comment by matthew-ballard** [comment]
@olepinard since this demo uses Sandbox data, what do you think about using the callouts for Sandbox data from the notebooks in that folder? E.g. https://github.com/planetlabs/notebooks/blob/master/jupyter-notebooks/workflows/planet_sandbox_data/land-surface-temperature/LST-climate-anomaly-mapping.ipynb

Also, what folder were you thinking of adding this to? the `planet_sandbox_data` folder?  I see right now it is in "ARPS" and I think that it would be better to not introduce a top-level folder for a specific data product

**Review by matthew-ballard — APPROVED** [review]
